[BUG] Invalid mangled folder names on Windows

**Describe the bug**
While building my project on Windows, I have the following error message:
```
CMake Error: Could not open file for write in copy operation K:/Gitlab/add_target/build/_deps/cmake_test-build/tests/tests/K:_Gitlab_add_target_build__deps_cmake_test-src_tests_cmake_test/add_section.cmake/src/CMakeLists.txt.tmp
CMake Error: : System Error: Invalid argument
CMake Error at build/_deps/cmake_test-src/cmake/cmake_test/add_dir.cmake:53 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  build/_deps/cmake_test-src/tests/CMakeLists.txt:15 (ct_add_dir)
```
The problem comes from the generated mangled path "K:/Gitlab/add_target/build/_deps/cmake_test-build/tests/tests/K:_Gitlab_add_target_build__deps_cmake_test-src_tests_cmake_test/add_section.cmake/src/CMakeLists.txt.tmp" which  failed at removing the colon which is forbidden. (see [here](https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names))

**To Reproduce**
You can try the project I am trying to test with cmake-test, on Windows:
```
git clone https://github.com/yobeonline/add_target.git --branch cmake-test
cmake -B build -S add_target
```

--- a/cmake/cmake_test/add_dir.cmake
+++ b/cmake/cmake_test/add_dir.cmake
@@ -48,6 +48,7 @@ function(ct_add_dir _ad_dir)
         file(RELATIVE_PATH _ad_rel_path "${_ad_abs_test_dir}" "${_ad_test_file}")
         file(TO_CMAKE_PATH "${_ad_abs_test_dir}" _ad_normalized_dir)
         string(REPLACE "/" "_" _ad_dir_prefix "${_ad_normalized_dir}")
+        string(REPLACE ":" "_" _ad_dir_prefix "${_ad_dir_prefix}")
 
         #Fill in boilerplate, copy to build dir
         configure_file(

